dbaccess/reachability.go

```diff
@@ -0,0 +1,75 @@
+package dbaccess
+
+import (
+	"github.com/kaspanet/kaspad/database2"
+	"github.com/kaspanet/kaspad/util/daghash"
+)
+
+var reachabilityDataBucket = database2.MakeBucket([]byte("reachability"))
+
+// ReachabilityDataCursor opens a cursor over all the
+// reachability data entries.
+func ReachabilityDataCursor(context Context) (database2.Cursor, error) {
+	accessor, err := context.accessor()
+	if err != nil {
+		return nil, err
+	}
+
+	return accessor.Cursor(reachabilityDataBucket.Path())
+}
+
+// StoreReachabilityData stores the reachability data of a block by its hash.
+func StoreReachabilityData(context Context, blockHash *daghash.Hash, reachabilityData []byte) error {
+	accessor, err := context.accessor()
+	if err != nil {
+		return err
+	}
+
+	key := reachabilityKey(blockHash)
+	return accessor.Put(key, reachabilityData)
+}
+
+// ClearReachabilityData clears the reachability data
+// from database.
+func ClearReachabilityData() error {
+	dbTx, err := NewTx()
+	if err != nil {
+		return err
+	}
+	defer dbTx.RollbackUnlessClosed()
+
+	accessor, err := dbTx.accessor()
+	if err != nil {
+		return err
+	}
+
+	// Collect all of the keys before deleting them. We do this
+	// as to not modify the cursor while we're still iterating
+	// over it.
+	keys := make([][]byte, 0)
+	cursor, err := accessor.Cursor(reachabilityDataBucket.Path())
```

Comment:
I think we want a general function that deletes all keys and a bucket. 
We had the same exact code in Stas' PR, so please change it there as well (can happen as a separate ticket if you prefer)